sdk/core/core/inc/az_span.h

```diff
@@ -135,161 +158,306 @@ AZ_NODISCARD az_span az_span_from_str(char* str);
+  memset(az_span_ptr(destination), fill, az_span_capacity(destination));
 }
 
 /**
- * @brief Set all the content of the span to @b fill without updating the length of the span.
- * This is util to set memory to zero before using span or make sure span is clean before use
+ * @brief az_span_copy copies the content of the source span to the destination span
  *
- * @param span source span
- * @param fill byte to use for filling span
+ * @param[in] destination The span whose bytes will be replaced by the source's bytes
+ * @param[in] source The span containing the bytes to copy to the destination
+ * @param[out] out_span A pointer to an az_span that receives the span referring to the destination
+ * span with its length updated
+ * @return An #az_result value indicating the result of the operation.
+ *          #AZ_OK if successful
+ *          #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY if the /p destination is not big enough to hold the
+ * source's length
  */
-AZ_INLINE void az_span_set(az_span span, uint8_t fill)
+AZ_NODISCARD az_result az_span_copy(az_span destination, az_span source, az_span* out_span);
+
+/**
+ * @brief az_span_copy_url_encode copies a URL in the source span to the destination span by
+ * url-encoding the source span's characters
+ *
+ * @param[in] destination The span whose bytes will receive the url-encoded source
+ * @param[in] source The span containing the non-url-encoded bytes
+ * @param[out] out_span A pointer to an az_span that receives the span referring to the
+ * destination span with its length updated
+ * @return An #az_result value indicating the result of the operation.
+ *          #AZ_OK if successful
+ *          #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY if the /p destination is not big enough to
+ * hold the source's length
+ */
+AZ_NODISCARD az_result
+az_span_copy_url_encode(az_span destination, az_span source, az_span* out_span);
+
+/******************************  SPAN PAIR  */
+
+/**
+ * An az_pair represents a key/value pair of az_span instances.
+ *  This is typically used for HTTP query parameters and headers.
+ */
+typedef struct
 {
-  memset(az_span_ptr(span), fill, az_span_capacity(span));
-}
+  az_span key;
+  az_span value;
+} az_pair;
 
 /**
- * @brief                  Appends one uint8_t value to the #az_span if there is enough capacity for
- * it.
- * @param[in]  span        #az_span where to append the char.
- * @param[in]  c           Value to be appended.
- * @param[out] out_span    #az_span returned as result of the concatenation.
- * @return                 An #az_result value indicating the result of the operation.
- *                         #AZ_OK                      If no errors occurs.
- *                         #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY   If there is not enough capacity left to append the `c`.
+ * @brief The AZ_PAIR_NULL macro returns an az_pair instance whose key and value fields are
+ * initialized to AZ_SPAN_NULL
```

Comment:
@danieljurek - is there a concept of "See also" or "refer to X" in doxygen and if so, how can we have links/pointers to other types/APIs in the doc comments?